tests/chainer_tests/functions_tests/array_tests/test_moveaxis.py

```diff
@@ -0,0 +1,168 @@
+
+    @attr.gpu
+    def test_forward_gpu(self):
+        self.check_forward(cuda.to_gpu(self.x))
+
+    def check_backward(self, x_data, g_data):
+        def f(x):
+            return functions.moveaxis(x, self.source, self.destination)
+
+        gradient_check.check_backward(
+            f, x_data, g_data, dtype='d', **self.check_backward_options)
+
+    def test_backward_cpu(self):
+        self.check_backward(self.x, self.g)
+
+    @attr.gpu
+    def test_backward_gpu(self):
+        self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.g))
+
+    def check_double_backward(self, x_data, g_data, gg_data):
+        def f(x):
+            y = functions.moveaxis(x, self.source, self.destination)
+            return y * y
+
+        gradient_check.check_double_backward(
+            f, x_data, g_data, gg_data, dtype='d',
+            **self.check_double_backward_options)
+
+    def test_double_backward_cpu(self):
+        self.check_double_backward(self.x, self.g, self.gg)
+
+    @attr.gpu
+    def test_double_backward_gpu(self):
+        self.check_double_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.g),
+                                   cuda.to_gpu(self.gg))
+
+
+@testing.parameterize(
+    {'source': 4, 'destination': 0},
+    {'source': 0, 'destination': 4},
+    {'source': 0, 'destination': -4},
+    {'source': -4, 'destination': 0},
+)
+class TestMoveaxisInvalidType(unittest.TestCase):
+
+    def setUp(self):
+        self.x = numpy.random.uniform(-1, 1, (2, 3, 4)).astype('f')
+
+    def check_type_error(self, x):
+        with self.assertRaises(type_check.InvalidType):
+            functions.moveaxis(x, self.source, self.destination)
+
+    def test_type_error_cpu(self):
+        self.check_type_error(self.x)
+
+    @attr.gpu
+    def test_type_error_gpu(self):
+        self.check_type_error(cuda.to_gpu(self.x))
+
+
+@testing.parameterize(
+    {'source': (1, 2), 'destination': (1, 2, 0)},
+    {'source': (0, 0), 'destination': (1, 2)},
+    {'source': (0, 1), 'destination': (2, 2)},
+)
+class TestMoveaxisValueError(unittest.TestCase):
+
+    def setUp(self):
+        self.x = numpy.random.uniform(-1, 1, (2, 3, 4)).astype('f')
+
+    def check_type_error(self, x):
+        with self.assertRaises(ValueError):
+            functions.moveaxis(x, self.source, self.destination)
+
+    def test_type_error_cpu(self):
+        self.check_type_error(self.x)
+
+    @attr.gpu
+    def test_type_error_gpu(self):
+        self.check_type_error(cuda.to_gpu(self.x))
+
+
+@testing.parameterize(
+    {'source': (1, 2), 'destination': (1, 2.0)},
+    {'source': (1, 2.0), 'destination': (1, 2)},
+)
+class TestMoveaxisTypeError(unittest.TestCase):
+
+    def setUp(self):
+        self.x = numpy.random.uniform(-1, 1, (2, 3, 4)).astype('f')
+
+    def check_type_error(self, x):
+        with self.assertRaises(TypeError):
+            functions.moveaxis(x, self.source, self.destination)
+
+    def test_type_error_cpu(self):
```

Comment:
`    @testing.with_requires('numpy!=1.11.*')` is fine.

Reply:
Thanks!
I didn'find, and It is numpy 1.11 bug.